clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -39,7 +39,42 @@
 
 import static org.apache.kafka.common.requests.ProduceResponse.INVALID_OFFSET;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class ProduceRequest extends AbstractRequest {
+    public static final Logger log = LoggerFactory.getLogger(ProduceRequest.class);
+
+    public static final String PRODUCE_REQUEST_PARSER_PROPERTY = "org.apache.kafka.common.requests.ProduceRequestParser";
+    public static final String PRODUCE_REQUEST_PARSER_ENV = "KAFKA_PRODUCE_REQUEST_PARSER";
+    public static final String PRODUCE_REQUEST_PARSER_DEFAULT = "org.apache.kafka.common.requests.DefaultProduceRequestParser";
+
+    private static ProduceRequestParser produceRequestParser = null;
+    static {
+        String produceRequestParserClassName = null;
+        try {
+            produceRequestParserClassName = System.getProperty(PRODUCE_REQUEST_PARSER_PROPERTY);
+
+            if (null != produceRequestParserClassName) {
+                log.debug("ProduceRequestParser class {} from system property {}", produceRequestParserClassName, PRODUCE_REQUEST_PARSER_PROPERTY);
+            } else {
+                produceRequestParserClassName = System.getenv(PRODUCE_REQUEST_PARSER_ENV);
```

Comment:
Majority of Kafka config doesn't use env variables. Please check `KafkaConfig.scala` if you want to add a config to broker side

Reply:
Checking for best way to get KafkaConfig into this component. Thank you.

Reply:
Is there an efficient way to use KafkaConfig in static context of a java class? I just scanned all the code and did not find a precedent. ProduceRequest.parse() is static, and all my changes are around delegating the behavior of that method to a separate class.

Reply:
Not that I'm aware of, also if you need this config to be dynamic we have DynamicBrokerConfig which depends on KafkaConfig.  

I would suggest that you open a KIP for discussion with a clear motivation for why you think this feature is important which is the most important part then the community can discuss the technical proposal of how would these config get loaded on the mailing list.

Reply:
Got it. I submitted a request to create an account. Will bring this up as soon it is created. Thank you for your guidance,